x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticationService.java

```diff
@@ -712,32 +710,23 @@ void finishAuthentication(User finalUser) {
          * successful
          */
         void writeAuthToContext(Authentication authentication) {
-            Runnable action = () -> {
-                logger.trace("Established authentication [{}] for request [{}]", authentication, request);
-                listener.onResponse(authentication);
-            };
             try {
                 authenticationSerializer.writeToContext(authentication, threadContext);
                 request.authenticationSuccess(authentication);
                 // Header for operator privileges will only be written if authentication actually happens,
                 // i.e. not read from either header or transient header
                 operatorPrivilegesService.maybeMarkOperatorUser(authentication, threadContext);
             } catch (Exception e) {
-                action = () -> {
-                    logger.debug(
+                logger.debug(
                         new ParameterizedMessage("Failed to store authentication [{}] for request [{}]", authentication, request), e);
-                    listener.onFailure(request.exceptionProcessingRequest(e, authenticationToken));
-                };
+                listener.onFailure(request.exceptionProcessingRequest(e, authenticationToken));
+                return;
             }
 
-            // we assign the listener call to an action to avoid calling the listener within a try block and auditing the wrong thing
-            // when an exception bubbles up even after successful authentication
-            action.run();
```

Comment:
Ah ok the same comment makes more sense here (because it has a try/catch block). As I suspected, it is no longer applicable even before this PR.